### ISSUE TITLE
Stress test fails on macOS (dyld: Library not loaded)

### ISSUE THREAD

**Issue by binarymaster** [adplug/adplug#138]
I have noticed it while adding GitHub Actions to the repository. Did several attempts in analysing and fixing it, but didn't came to any solution.

- Created a branch [`macos`](https://github.com/adplug/adplug/commits/macos) for it.
- In the master branch there is a workaround applied: 3a90fc3d.

The output is:
```sh
dyld: Library not loaded: /usr/local/lib/libadplug-2.3.4-beta.0.dylib
  Referenced from: /Users/runner/work/adplug/adplug/test/.libs/stresstest
  Reason: image not found
Testing: i-85.bmf - [FAIL] (killed by signal 6)
```

Stack trace:
```
* thread https://github.com/adplug/adplug/issues/1, stop reason = signal SIGSTOP
  * frame #0: 0x0000000113ec3e0a dyld`__abort_with_payload + 10
    frame https://github.com/adplug/adplug/issues/1: 0x0000000113eecbb1 dyld`abort_with_payload_wrapper_internal + 80
    frame https://github.com/adplug/adplug/issues/2: 0x0000000113eecbe3 dyld`abort_with_payload + 9
    frame https://github.com/adplug/adplug/pull/3: 0x0000000113e6c412 dyld`dyld::halt(char const*) + 672
    frame https://github.com/adplug/adplug/issues/4: 0x0000000113e70383 dyld`dyld::_main(macho_header const*, unsigned long, int, char const**, char const**, char const**, unsigned long*) + 9932
    frame https://github.com/adplug/adplug/pull/5: 0x0000000113e68224 dyld`dyldbootstrap::start(dyld3::MachOLoaded const*, int, char const**, dyld3::MachOLoaded const*, unsigned long*) + 450
    frame https://github.com/adplug/adplug/issues/6: 0x0000000113e68025 dyld`_dyld_start + 37
```

Cc @miller-alex as the test creator.

**Comment by miller-alex** [comment]
I identified the problem (System Integrity Protection vs DYLD_LIBRARY_PATH) and added a few commits to the macos branch to address it. When you're satisfied with the result you can add the fix to master.

(Maybe I should get rid of system() entirely. Initially it was a simple and portable solution, but it's becoming messy.)

BTW, the (ubuntu-latest, clang) builds seem to be broken currently ("C compiler cannot create executables"), but that's obviously unrelated.

**Comment by binarymaster** [comment]
> BTW, the (ubuntu-latest, clang) builds seem to be broken currently ("C compiler cannot create executables"), but that's obviously unrelated.

Indeed, so I have checked `config.log` output:
```
clang: error while loading shared libraries: libtinfo.so.5: cannot open shared object file: No such file or directory
```
Adding `libncurses5` package fixes that. But it's strange that recently it worked just fine, perhaps something has changed in runner's environment.

Addressed in 344308cf.

**Comment by binarymaster** [comment]
It's added now, many thanks!
